auth/realm/ldap/src/main/java/org/wildfly/security/auth/realm/ldap/X509EvidenceVerifier.java

```diff
@@ -217,11 +217,8 @@ public boolean verifyCertificate(X509Certificate certificate, Attributes attribu
             try {
                 for (int i = 0; i < size; i++) {
                     Object attrCertificate = attribute.get(i);
-                    if (attrCertificate != null){
-                        if (MessageDigest.isEqual(certificate.getEncoded(), (byte[]) attrCertificate)) {
-                            return true;
-                        }
-                    }
+                    return attrCertificate != null
```

Comment:
We need to be a bit careful here. We still need an if statement to make sure we only return true if `attrCertificate != null` and `MessageDigest.isEqual(certificate.getEncoded(), (byte[]) attrCertificate)` are both `true`. If not, the current changes return `false` but that's different from the current behaviour.